thanks!  

Not running on appveyor, maybe it needs `\r` instead of `\n`?

```
[  ERROR   ] 2 errors, listed below:
[  ERROR   ] test/functional\terminal\scrollback_spec.lua @ 397: 'scrollback' option set to 0 behaves as 1
test\functional\ui\screen.lua:587: Failed to match any screen lines.
Expected (anywhere): "30: line                      "
Actual:
  |(c) 2016 Microsoft Corporation|
  |. All rights reserved.        |
  |                              |
  |C:\projects\neovim>C:/projects|
  |/neovim/build/bin/shell-test R|
  |EP 31 line{1: }                   |
  |{3:-- TERMINAL --}                |
stack traceback:
	test\functional\ui\screen.lua:587: in function '_wait'
	test\functional\ui\screen.lua:370: in function 'expect'
	test/functional\terminal\scrollback_spec.lua:407: in function <test/functional\terminal\scrollback_spec.lua:397>
[  ERROR   ] test/functional\terminal\scrollback_spec.lua @ 413: 'scrollback' option deletes lines (only) if necessary
test\functional\ui\screen.lua:587: Failed to match any screen lines.
Expected (anywhere): "30: line                      "
Actual:
  |0.14393]                      |
  |(c) 2016 Microsoft Corporation|
  |. All rights reserved.        |
  |                              |
  |$C:/projects/neovim/build/bin/|
  |shell-test REP 31 line{1: }       |
  |{3:-- TERMINAL --}                |
stack traceback:
	test\functional\ui\screen.lua:587: in function '_wait'
	test\functional\ui\screen.lua:370: in function 'expect'
	test/functional\terminal\scrollback_spec.lua:428: in function <test/functional\terminal\scrollback_spec.lua:413>
```

Also the macOS hang has a SIGSTOP backtrace (does that mean CI stopped it because of a hang? or did it crash, and then it "hangs" in interactive gdb?)

```
(lldb) target create "build/bin/nvim" --core "/cores/core.13683"
Traceback (most recent call last):
(lldb) bt all
* thread #1, stop reason = signal SIGSTOP
  * frame #0: 0x00007fff581de5d1 libsystem_platform.dylib`_platform_memmove$VARIANT$Ivybridge + 49
    frame #1: 0x000000010e0ff49b nvim`ml_append_int(buf=0x00007fc17900d600, lnum=-3, line="                                                                       ", len=9, newfile=false, mark=0) at memline.c:2046:7
    frame #2: 0x000000010e0fd2e0 nvim`ml_append(lnum=-3, line="                                                                       ", len=0, newfile=false) at memline.c:1916:10
    frame #3: 0x000000010e24532f nvim`refresh_scrollback(term=0x00007fc178818800, buf=0x00007fc17900d600) at terminal.c:1258:5
    frame #4: 0x000000010e241ea5 nvim`refresh_terminal(term=0x00007fc178818800) at terminal.c:1169:3
    frame #5: 0x000000010e24516b nvim`refresh_timer_cb(watcher=0x000000010e410160, data=0x0000000000000000) at terminal.c:1187:3
    frame #6: 0x000000010e0413be nvim`time_event(argv=0x00007ffee1ca8418) at time.c:50:3
    frame #7: 0x000000010e03c812 nvim`multiqueue_process_events(this=0x00007fc178405600) at multiqueue.c:150:7
    frame #8: 0x000000010e086492 nvim`command_line_execute(state=0x00007ffee1ca8798, key=-26365) at ex_getln.c:543:7
    frame #9: 0x000000010e2209bb nvim`state_enter(s=0x00007ffee1ca8798) at state.c:71:26
    frame #10: 0x000000010e07dac8 nvim`command_line_enter(firstc=58, count=1, indent=0) at ex_getln.c:411:3
    frame #11: 0x000000010e07d020 nvim`getcmdline(firstc=58, count=1, indent=0) at ex_getln.c:2000:20
    frame #12: 0x000000010e07e45a nvim`getexline(c=58, cookie=0x0000000000000000, indent=0) at ex_getln.c:2178:10
    frame #13: 0x000000010e05cc73 nvim`do_cmdline(cmdline=0x0000000000000000, fgetline=(nvim`getexline at ex_getln.c:2173), cookie=0x0000000000000000, flags=0) at ex_docmd.c:513:47
    frame #14: 0x000000010e12ef3a nvim`nv_colon(cap=0x00007ffee1ca9268) at normal.c:4599:18
    frame #15: 0x000000010e12cbb0 nvim`normal_execute(state=0x00007ffee1ca91e0, key=58) at normal.c:1131:3
    frame #16: 0x000000010e2209bb nvim`state_enter(s=0x00007ffee1ca91e0) at state.c:71:26
    frame #17: 0x000000010e123e07 nvim`normal_enter(cmdwin=false, noexmode=false) at normal.c:463:3
    frame #18: 0x000000010e0dec30 nvim`main(argc=9, argv=0x00007ffee1ca9548) at main.c:569:3
    frame #19: 0x00007fff57ff93d5 libdyld.dylib`start + 1
    frame #20: 0x00007fff57ff93d5 libdyld.dylib`start + 1
(lldb) 

No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```

--- a/test/functional/terminal/scrollback_spec.lua
+++ b/test/functional/terminal/scrollback_spec.lua
@@ -403,12 +403,8 @@ describe("'scrollback' option", function()
     end
 
     curbufmeths.set_option('scrollback', 0)
-    if iswin() then
-      feed_data('for /L %I in (1,1,30) do @(echo line%I)\r')
-    else
-      feed_data('awk "BEGIN{for(n=1;n<=30;n++) print \\\"line\\\" n}"\n')
-    end
-    screen:expect{any='line30                        '}
+    feed_data(nvim_dir..'/shell-test REP 31 line'..(iswin() and '\r' or '\n'))
+    screen:expect{any='30: line                      '}
     retry(nil, nil, function() expect_lines(7) end)
 
     screen:detach()
@@ -428,13 +424,8 @@ describe("'scrollback' option", function()
     -- Wait for prompt.
     screen:expect{any='%$'}
 
-    if iswin() then
-      feed_data('for /L %I in (1,1,30) do @(echo line%I)\r')
-    else
-      feed_data('awk "BEGIN{for(n=1;n<=30;n++) print \\\"line\\\" n}"\n')
-    end
-
-    screen:expect{any='line30                        '}
+    feed_data(nvim_dir.."/shell-test REP 31 line"..(iswin() and '\r' or '\n'))
+    screen:expect{any='30: line                      '}
 
     retry(nil, nil, function() expect_lines(33, 2) end)
     curbufmeths.set_option('scrollback', 10)
@@ -445,18 +436,14 @@ describe("'scrollback' option", function()
     -- Terminal job data is received asynchronously, may happen before the
     -- 'scrollback' option is synchronized with the internal sb_buffer.
     command('sleep 100m')
-    if iswin() then
-      feed_data('for /L %I in (1,1,40) do @(echo line%I)\r')
-    else
-      feed_data('awk "BEGIN{for(n=1;n<=40;n++) print \\\"line\\\" n}"\n')
-    end
 
-    screen:expect{any='line40                        '}
+    feed_data(nvim_dir.."/shell-test REP 41 line"..(iswin() and '\r' or '\n'))
+    screen:expect{any='40: line                      '}
 
     retry(nil, nil, function() expect_lines(58) end)
     -- Verify off-screen state
-    eq((iswin() and 'line36' or 'line35'), eval("getline(line('w0') - 1)"))
-    eq((iswin() and 'line27' or 'line26'), eval("getline(line('w0') - 10)"))
+    eq((iswin() and '36: line' or '35: line'), eval("getline(line('w0') - 1)"))
+    eq((iswin() and '27: line' or '26: line'), eval("getline(line('w0') - 10)"))
 
     screen:detach()
   end)

